omod/src/main/java/org/openmrs/module/coreapps/page/controller/systemadministration/SystemAdministrationPageController.java

```diff
@@ -17,6 +18,11 @@ public void controller(PageModel model, UiSessionContext emrContext,
                            @SpringBean("appFrameworkService") AppFrameworkService appFrameworkService) {
 
         emrContext.requireAuthentication();
+		
+		if (!emrContext.getCurrentUser().getPrivileges().toString().contains("coreapps.systemAdministration")
```

Comment:
What about using `emrContext.getCurrentUser().hasPrivilege("coreapps.systemAdministration")` for the privilege check?

Reply:
I just tried to make that change and was unable to get it to work properly using that.

Reply:
Did you get an error or did it still allow low-privilege users to access the page?